src/init.cpp

```diff
@@ -175,6 +175,8 @@ static fs::path GetPidFile(const ArgsManager& args)
 
 [[nodiscard]] static bool CreatePidFile(const ArgsManager& args)
 {
+    if (args.IsArgNegated("-pid")) return true;
```

Comment:
it's hard to review 14 commits, as hinted before, could the fix be in the same commit as the test - I personally am lacking a lot of context and wouldn't want this commit to be merged without making sure a test is added as well

Reply:
PR does not contain a test for this as I thought it was straightforward enough to skip and I didn't want to add more things to review/maintain. If we pass `-nopid` we simply do not:
- Create the file
- Write the PID
- Set the `bool` to remember deleting the file
- (Delete the file)

https://github.com/bitcoin/bitcoin/blob/95a0104f2e9869799db84add108ae8c57b56d360/src/init.cpp#L165-L192